Спринт 9/29 "Стили, Nемы, Context, Intent, EditText, TextWatcher, хранение данных"

philipmaslov opened the pull request:
Добавил стили и темы

Commit:
Спринт 9/29 → Тема 1/5: Стили и темы → Урок 3/3
Добавил стили и темы

Commit:
Ссылка на урок: [Спринт 9/29 → Тема 3/5: EditText и TextWatcher → Урок 3/3: Домашнее задание]
- Экран Настройки (добавлено поведение при нажатии на кнопки)
- Экран Поиск (реализован ввод текста)

Дополнительно:
Переверстал заголовок и футер экранов (добавил <- и убрал футер)

Commit:
Ссылка на урок: [Спринт 9/29 → Тема 3/5: EditText и TextWatcher → Урок 3/3: Домашнее задание]
- добавил сохранение значения строки поиска

Commit:
- исправление ошибок
- добавил обработку нажатия кнопки "<-"

ArturNurtdinov commented on app/src/main/java/com/example/playlistmaker/MainActivity.kt:
```diff
@@ -24,14 +22,14 @@ class MainActivity : AppCompatActivity() {
-            val libraryIntent = Intent(this, NothingActivity::class.java)
-            startActivity(libraryIntent)
-        }
+//
```
Закоментированный код лучше сразу удалять. Хранить закоментированный код в проекте - плохая практика, в случае чего этот код можно восстановить с помощью истории гита

ArturNurtdinov commented on app/src/main/java/com/example/playlistmaker/SearchActivity.kt:
```diff
@@ -1,20 +1,86 @@
+
+    companion object {
+        const val SEARCH_INPUT = "SEARCH_INPUT"
+        const val SEARCH_INPUT_DEF = ""
```
Константы можно сделать приватными, так как они относятся только к логике работы данного класса

ArturNurtdinov commented on app/src/main/java/com/example/playlistmaker/SettingsActivity.kt:
```diff
@@ -16,5 +24,63 @@ class SettingsActivity : AppCompatActivity() {
+                startActivity(
+                    Intent.createChooser(
+                        supportRequest,
+                        "Отправка запроса в службу сопровождения"
```
Данную строчку тоже необходимо вынести в ресурсы приложения strings.xml, чтобы работала локализация

ArturNurtdinov commented on app/src/main/java/com/example/playlistmaker/SettingsActivity.kt:
```diff
@@ -16,5 +24,63 @@ class SettingsActivity : AppCompatActivity() {
+                setData(Uri.parse(getString(R.string.yandex_offer)))
+            }
+            try {
+                startActivity(Intent.createChooser(userPolicy, "Просмотр web-страницы"))
```
И эту строчку тоже нужно вынести в strings.xml

Commit:
Исправления по замечаниям на review:
- Экран "Поиск" - добавил скрытие клавиатуры и фокуса с поля поиска при тапе на "x"
- Экран "Настройки" > "Написать в поддержку" - Исправил поведение, теперь e-mail заполняется корректно
- Перенёс строки в ресурсы
- Удалил закомментированный код
- мелкие испралвения

Commit:
Исправления по замечаниям на review:
- Экран "Поиск" - строка поиска перевёрстана с FrameLayout

Commit:
Исправления по замечаниям на review:
- мелкие правки

philipmaslov commented:
@ArturNurtdinov привет, спасибо за комментарии. 
Внёс исправления. Посмотри, пожалуйста.

ArturNurtdinov commented:
> @ArturNurtdinov привет, спасибо за комментарии. Внёс исправления. Посмотри, пожалуйста.

Привет @philipmaslov. Кажется, ты не запушил изменения - в PR нет коммитов после моего ревью

philipmaslov commented:
> > @ArturNurtdinov привет, спасибо за комментарии. Внёс исправления. Посмотри, пожалуйста.
> 
> Привет @philipmaslov. Кажется, ты не запушил изменения - в PR нет коммитов после моего ревью
@ArturNurtdinov да, верно. Подлил изменения.

ArturNurtdinov commented on app/src/main/res/layout/activity_search.xml:
```diff
@@ -28,34 +32,26 @@
+            android:paddingStart="14dp">
 
             <EditText
                 android:id="@+id/input_edit_text"
```
Этой вью еще нужен paddingEnd добавить, чтобы длинный текст не залезал на крестик

philipmaslov replied:
@ArturNurtdinov спасибо, что обратил внимание. Крестик по умолчанию скрыт и появляется только после того, как установлен фокус и введен хотя бы один символ. 

В последнем комите поправил, установил android:visibility="gone". Сейчас работает корректно, текст не перекрывает элемент.

![telegram-cloud-photo-size-2-5368670994031571607-y](https://github.com/user-attachments/assets/e2cc4af9-376a-4e92-b6a3-d469c8bc800b)

Commit:
Исправления по замечаниям на review:
- для экрана "Поиск" добавил для "х" Android:visibility="gone"
- для экрана "Настройки" - переписал функцию отправки e-mail в соответствии с https://developer.android.com/guide/components/intents-common#Email

philipmaslov commented:
@ArturNurtdinov хотел отдельно поблагодарить за комментарий с email (упаковка email поулчателя в массив). Обсудили с наставником, посмотрели документацию https://developer.android.com/guide/components/intents-common#Email. Переписал код, в соответствии с требованиями. :)

philipmaslov commented:
@ArturNurtdinov запушил все изменения, отправил на фильнальный ревью.
